### DISCUSSION TITLE
Cannot open pdf file using custome scheme handler

### DISCUSSION THREAD

**Discussion by abid76** [cefsharp/CefSharp#4261] [Q&A] [answered]
This is a duplicate of https://stackoverflow.com/questions/73906457/cefsharp-doesnt-display-local-pdf-file-after-upgrade-to-105-3-390

Setup:
OS: Windows 11
.NET Version: .NET Framework 4.8
Architecture: x86

After upgrading from 79.1.360 to 105.3.390 I cannot open pdf files when using a custom scheme handler. When trying to open a pdf file using "local://path/to/file.pdf" Windows OS opens a window saying: _You'll need a new app to open this chrome-extension link_.

When using _file_ scheme everything works fine. Furthermore, I checked out the tag v105.3.390 and added my custom scheme handler to _CefSharp.Wpf.Example_: everything works fine.

I have absolutely no idea what's going on and how to solve this issue ... any help is appreciated!

**Comment by abid76** [comment, marked as answer]
Found it out myself. When opening a PDF file, an additional URL is processed: _chrome-extension://mhjfbmdgcfjbbpaeojofohoefgiehjai/index.html_ (**anybody knows why?**).

I was using a `RequestHandler` which delegates all none local URLs to the default OS browser. This causes the OS to open the window 

> You'll need a new app to open this chrome-extension link.

See all details on [stackoverflow](https://stackoverflow.com/a/73931959/9826710).

**Reply by amaitland** [reply]
> When opening a PDF file, an additional URL is processed: _chrome-extension://mhjfbmdgcfjbbpaeojofohoefgiehjai/index.html_ (**anybody knows why?**).

This is expected. The `PDF` viewer migrated from being a `PPAPI Plugin` to being an internal extension a while back.

https://source.chromium.org/search?q=mhjfbmdgcfjbbpaeojofohoefgiehjai&ss=chromium
